packages/web3-eth2-core/src/index.ts

```diff
@@ -0,0 +1,89 @@
+import Axios, {AxiosInstance} from 'axios'
+
+import { ETH2BaseOpts, ETH2Function } from '../types/index'
+import { IBaseAPISchema } from './schema'
+
+export class ETH2Core {
+    private _httpClient: AxiosInstance
+
+    [ key: string ]: ETH2Function | any;
+    
+    name: string
+    provider: string
+    protectProvider: boolean
+
+    constructor(provider: string, schema: IBaseAPISchema, opts: ETH2BaseOpts = {}) {
+        this.name = schema.packageName
+        this.setProvider(`${provider}${schema.routePrefix}`)
+        this.protectProvider = opts.protectProvider || false
+        this.buildAPIWrappersFromSchema(schema)
+    }
+
+    static createHttpClient(baseUrl: string): AxiosInstance {
+        try {
+            return Axios.create({
+                baseURL: baseUrl
+            })
+        } catch (error) {
+            throw new Error(`Failed to create HTTP client: ${error.message}`)
+        }
+    }
+
+    setProvider(provider: string) {
+        try {
+            if (!provider || typeof provider !== 'string' || !/^http(s)?:\/\//i.test(provider)) {
```

Comment:
Do you want to add a check for `this.protectProvider` here?

Reply:
So `protectProvider` is for the global instance of `web3` to check if it should overwrite the provider for a particular sub package when using the future `.use` functionality. So if a user was to do:

```javascript
const web3 = new Web3(providerURL)
web3.use(Eth2Beaconchain(eth2ProviderURL, schema, { protectProvider: true }))

// Because the base web3 instance is probably going to use a different provider than the ETH2 packages,
// we want the use to have a control to dictate whether a certain subpackage should be
// affected by ther global use of:
web3.setProvider(newProviderURL)

// In this code example, web3.provider = newProviderURL
// while web3.eth2Beaconchain.provider = eth2ProviderURL
```

Reply:
I did accidentally leave the call to `super` in the `ETH2BeaconChain` class as:

`super(provider, schema, { protectProvider: true })` instead of `super(provider, schema, opts)` where `opts` is passed by the user - will commit the changes for this soon